js/pathfinding.js

```diff
@@ -149,8 +137,14 @@ function calcNoDiagonals(waypoints) {
 	return diagonals;
 }
 
+/**
+ * Estimate the travel distance between two points, as the crow flies. Most of the time, this is 1
+ * per space, but for a square grid using 5-10-5 diagonals, count each diagonal as an extra 0.5
+ */
 function estimateCost(pos, target) {
-	return Math.max(Math.abs(pos.x - target.x), Math.abs(pos.y - target.y));
+	const distX = Math.abs(pos.x - target.x);
+	const distY = Math.abs(pos.y - target.y);
+	return Math.max(distX, distY) + (use5105 ? Math.min(distX, distY) * 0.5 : 0);
```

Comment:
This logic is flawed. For a single diagonal step on 5/10/5 this will estimate the cost to be 1.5, while the actual distance is 1. For A* to work correctly it's important that the estimated cost is always equal or cheaper to the actual cost, which isn't the case here. Options would be to either revert this change or modifying the new logic to ensure that it cannot ever overestimate the distance.

Reply:
The way I've changed the logic in `getNode` means that a single diagonal step actually does cost `1.5`, not either `1.0` or `2.0`. The difference is important for the algorithm - it is better to be in a space where the next diagonal will cost 5 ft. of movement (i.e. in the algorithm the cost is `x.0`) than to be in the same space where the next diagonal will cost 10 ft. (i.e. in the algorithm the cost is `x.5`), and that is reflected in the `.5` being part of both the cost and estimated cost.

Even at the end of the path, the difference _could_ be important (if you want to place a waypoint and carry on moving). Only when you're finished moving does the difference between `x.5` and `x.0` stop mattering.

Reply:
Yes, you're right.